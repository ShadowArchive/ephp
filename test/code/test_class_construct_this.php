<?php

class C {
    public $a = 200;
    function __construct() {
        print "a => " . $this->a . "\n";
    }
}

class A {
    public $a = 100;
    function __construct() {
        print "a => " . $this->a . "\n";
    }
}

class B extends A {
    function __construct() {
        print "from ";
        parent::__construct();
        print "end\n";

        print "from ";
        A::__construct();
        print "end\n";
    }
}

class D {
    function __construct() {
        C::__construct();
    }
}

$b = new B;
$d = new D;

