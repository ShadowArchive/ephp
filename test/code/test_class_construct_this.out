from a => 100
end
from a => 100
end

Fatal error: Non-static method C::__construct() cannot be called statically, assuming $this from incompatible context in {{CWD}}/test/code/test_class_construct_this.php on line 31
